build.gradle

```diff
@@ -7,13 +7,19 @@ plugins {
 	id 'maven-publish'
 }
 
+group = 'com.iexec.blockchain'
+sourceCompatibility = '11'
+targetCompatibility = '11'
+
 ext {
 	openFeignVersion = '11.6'
+	gitBranch = 'git rev-parse --abbrev-ref HEAD'.execute().text.trim()
 }
 
-group = 'com.iexec.blockchain'
-sourceCompatibility = '11'
-targetCompatibility = '11'
+if (gitBranch != 'main' && gitBranch != 'master' && ! (gitBranch ==~ '(release|hotfix|support)/.*')) {
+	version            += '-NEXT-SNAPSHOT'
+	iexecCommonVersion += '-NEXT-SNAPSHOT'
```

Comment:
Shouldn't we try and load the last `iexec-common` next snapshot version instead of a fixed one? Based on what we currently have, I think NS versions can be removed from our repository. This means a project can suddenly not compile anymore :(

I think of 2 solutions:
1. If we're not in one of the "master" branches, we automatically fetch the last NS version ;
2. If `iexecCommonVersion` is not set in `gradle.properties` - or set to something like `dev`, we automatically fetch the last NS version.

Any thoughts about this?

Reply:
In my opinion, the developer must manage the base version.
It must not change through a programmatic behavior and fetching the last NS version brings its own troubles:
What will happen when the last NS version is several versions after the one written in `gradle.properties` ?
What will happen when a NS version has breaking changes ?

I prefer the following suggestions which will be easier to implement:
* `iexecCommonVersion` stays outside of this dynamic versioning
* The latest `iexecCommonVersion` is used as much as possible in `gradle.properties` of other modules (it alleviates the point you raise and we try as much as possible to use the latest version everywhere)
* All versions are aligned across all modules and we manipulate a single global version for the platform

What do you think ?

Reply:
I like the second suggestion. But that means we need to update all projects when a new version of `iexec-common` is released... Which takes some time.
Could that be done using a Github bot?

Reply:
`iexec-common` removed from dynamic versioning